### PR TITLE
 Fix #8305: failed to generate the changelog between HLC and Modular

### LINKED ISSUE #8305: Support the breaking detection between HLC and modular classical layer
failed to generate the changelog between HLC and Modular:
image

it's a issue in https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/src/hlc/utils/automaticGenerateChangeLogAndBumpVersion.ts#L28
where caused this error.
From HLC package, we have src/models/parameters.ts to generate the api version and let the tool read this to check whether the release is a stable release. but in modular, src/models/parameters.ts doesn't exist.

steps to repro:
1. checkout https://github.com/Azure/azure-sdk-for-js/pull/29770
2. in JS sdk repo root path, run ``` changelog-tool sdk/networkanalytics/arm-networkanalytics```

### PR DESCRIPTION
**Pull request by wanlwanl**
Generated CHANGELOG.md (2024-05-24):

HLC
```
# Release History
    
## 4.0.0 (2024-05-24)
    
**Breaking Changes**

  - Interface SuppressionsListOptionalParams no longer has parameter skipToken
    
## 3.2.0 (2023-08-03)

**Features**

- Class AdvisorManagementClient has a new signature
- Interface RecommendationsListNextOptionalParams no longer has parameter filter
- Interface RecommendationsListNextOptionalParams no longer has parameter skipToken
- Interface RecommendationsListNextOptionalParams no longer has parameter top
- Interface SuppressionsListNextOptionalParams no longer has parameter skipToken
- Interface SuppressionsListNextOptionalParams no longer has parameter top

## 3.1.0 (2022-11-14)

**Features**

- Added Interface ConfigData
- Added Interface ResourceRecommendationBase
- Added Interface SuppressionContract

## 3.0.3 (2022-10-08)

**Bugs Fixed**

- revert credential scopes

## 3.0.2 (2022-09-30)

**Bugs Fixed**

- fix better user experience of credential scopes in government cloud

## 3.0.1 (2022-03-22)

**features**

- bug fix

## 3.0.0 (2021-12-29)

The package of @azure/arm-advisor is using our next generation design principles since version 3.0.0, which contains breaking changes.

To understand the detail of the change, please refer to [Changelog](https://aka.ms/js-track2-changelog).

To migrate the existing applications to the latest version, please refer to [Migration Guide](https://aka.ms/js-track2-migration-guide).

To learn more, please refer to our documentation [Quick Start](https://aka.ms/azsdk/js/mgmt/quickstart).

```

Modular

> Use sdk from failed PR: https://github.com/Azure/azure-sdk-for-js/pull/29770

```
# Release History
    
## 2.0.0 (2024-05-24)
    
**Features**

  - Added Interface ArmOperationStatus
  - Added Interface DataProductsCatalogsOperations
  - Added Interface DataProductsOperations
  - Added Interface DataTypesOperations
  - Added Interface NetworkAnalyticsClientOptions
  - Added Interface OperationsOperations
  - Added Interface PagedAsyncIterableIterator
  - Added Interface PagedOperation
  - Added Interface PageSettings
  - Added Interface RestorePollerOptions
  - Added Class NetworkAnalyticsClient
  - Added Type Alias ContinuablePage
  - Added Type Alias ResourceProvisioningState
  - Added function restorePoller

**Breaking Changes**

  - Removed operation group DataProducts
  - Removed operation group DataProductsCatalogs
  - Removed operation group DataTypes
  - Removed operation group Operations
  - Deleted Class MicrosoftNetworkAnalytics
  - Interface DataProductsCreateOptionalParams no longer has parameter resumeFrom
  - Interface DataProductsDeleteOptionalParams no longer has parameter resumeFrom
  - Interface DataProductsUpdateOptionalParams no longer has parameter resumeFrom
  - Interface DataTypesCreateOptionalParams no longer has parameter resumeFrom
  - Interface DataTypesDeleteDataOptionalParams no longer has parameter resumeFrom
  - Interface DataTypesDeleteOptionalParams no longer has parameter resumeFrom
  - Interface DataTypesUpdateOptionalParams no longer has parameter resumeFrom
  - Type of parameter tags of interface DataProductUpdate is changed from {
        [propertyName: string]: string;
    } to Record<string, string>
  - Type of parameter info of interface ErrorAdditionalInfo is changed from Record<string, unknown> to Record<string, any>
  - Type of parameter userAssignedIdentities of interface ManagedServiceIdentity is changed from {
        [propertyName: string]: UserAssignedIdentity;
    } to Record<string, UserAssignedIdentity>
  - Type of parameter tags of interface TrackedResource is changed from {
        [propertyName: string]: string;
    } to Record<string, string>
  - Removed Enum KnownActionType
  - Removed Enum KnownBypass
  - Removed Enum KnownControlState
  - Removed Enum KnownCreatedByType
  - Removed Enum KnownDataProductUserRole
  - Removed Enum KnownDataTypeState
  - Removed Enum KnownDefaultAction
  - Removed Enum KnownManagedServiceIdentityType
  - Removed Enum KnownOrigin
  - Removed Enum KnownProvisioningState
  - Removed Enum KnownVersions
  - Removed function getContinuationToken
    
    
## 1.0.0 (2024-01-24)

The package of @azure/arm-networkanalytics is using our next generation design principles. To learn more, please refer to our documentation [Quick Start](https://aka.ms/azsdk/js/mgmt/quickstart).

```

### CODE REVIEW

File: tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+const getApiVersionTypeFromRestClient: IApiVersionTypeExtractor = (packageRoot: string): ApiVersionType => {
+    const clientPath = findRestClientPath(packageRoot);
+    const apiVersion = findApiVersionInRestClient(clientPath);
+    if (apiVersion && apiVersion.indexOf("-preview") >= 0) return ApiVersionType.Preview;
+    if (apiVersion && apiVersion.indexOf("-preview") < 0) return ApiVersionType.Stable;
+    return ApiVersionType.None;
+};
+
+// TODO: not implemented: need a example
```
Comment (wanlwanl) [inline tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts:54]: Not implemented yet.

Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts]: done

File: tools/js-sdk-release-tools/src/llc/utils/generateChangelog.ts [hunk 2]
```diff
@@ -60,8 +61,9 @@ export async function generateChangelog(packagePath) {
                 logger.logWarn("The latest package released in NPM doesn't contain review folder, so generate changelog same as first release");
                 generateChangelogForFirstRelease(packagePath, version);
             } else {
-                let apiMdFileNPM = path.join(tempReviewFolder, fs.readdirSync(tempReviewFolder)[0]);
-                let apiMdFileLocal = path.join(packagePath, 'review', fs.readdirSync(path.join(packagePath, 'review'))[0]);
+                const npmPackageRoot = path.join(packagePath, 'changelog-temp', 'package');
+                const apiMdFileNPM = getApiReviewPath(npmPackageRoot);
+                const apiMdFileLocal = getApiReviewPath(packagePath);
```
Comment (qiaozha) [inline tools/js-sdk-release-tools/src/llc/utils/generateChangelog.ts:66]: maybe not a high priority for now but should we error out if it's comparing between RLC and HLC? or Modular api layer and HLC ?

**Review by qiaozha — APPROVED** [review]
LGTM.

File: tools/js-sdk-release-tools/src/common/utils.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const reviewDir = path.join(packageRoot, 'review');
+    switch (sdkType) {
+        case SDKType.MLC:
+            const fileNames = fs.readdirSync(reviewDir).sort();
+            if (fileNames.length === 0) {
+                logger.logError(`Expects 1 API report, but find nothing.`);
+                process.exit(1);
+            }
+            // TODO: use a more concrete rule to find xxx.api.md
+            return path.join(packageRoot, 'review', fileNames[fileNames.length - 1]);
```
[the remark this answers is not in the document]
Comment (qiaozha) [reply tools/js-sdk-release-tools/src/common/utils.ts]: Thanks for the investigation. I am a little doubt about the sort order actually, different os may have different behavior ? not sure.

**Pull request merged**